### PR TITLE
Update documentation for Datastream Override Feature

### PR DESCRIPTION
**Pull request by addb**
## Description
- Added short description for datastream overrides feature to the sendEvent API reference.
- Added Swift samples. 
- Added ObjectiveC sample
- Update ExperienceEvent class structure
- Updated Event reference with the new fields `datastreamIdOverride` and `datastreamConfigOverride`

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: Documentation/api-reference.md [hunk 1]
```diff
@@ -100,13 +100,17 @@ See [MobileCore.resetIdentities](https://developer.adobe.com/client-sdks/documen
 
 Sends an Experience event to Edge Network.
 
+The sendEvent API has been improved to support Datastream overrides. This allows you to adjust your datastreams without the need for new ones or modifications to existing settings. The process involves two steps:
```
Comment (emdobrin) [inline Documentation/api-reference.md:103]: rewording suggestion:
The sendEvent API has been improved to support Datastream overrides. -> Starting with vx.y.z the sentEvent API supports optional Datastream overrides.

File: Documentation/api-reference.md [hunk 2]
```diff
@@ -128,6 +132,69 @@ Edge.sendEvent(experienceEvent: experienceEvent) { (handles: [EdgeEventHandle])
...
+```swift
+// Create Experience event from dictionary
+var xdmData : [String: Any] = ["eventType" : "SampleXDMEvent",
+                              "sample": "data"]
+
+ let configOverrides: [String: Any] = [
+                                        "com_adobe_experience_platform": [
+                                          "datasets": [
+                                            "event": [
+                                              "datasetId": "eventDatasetIdOverride"
```
Comment (emdobrin) [inline Documentation/api-reference.md:163]: for consistency can we use:
SampleEventDatasetIdOverride
SampleProfileDatasetIdOverride
SamplePropertyToken

File: Documentation/event-reference.md [hunk 3]
```diff
@@ -89,10 +89,12 @@ If the required `xdm` key is not present in the event data payload, the event is
 | --- | ---------- | -------- | ----------- |
 | xdm | <code>[String:&nbsp;Any]</code> | Yes | XDM formatted data; use an `XDMSchema` implementation for better XDM data ingestion and data format control. |
 | data | <code>[String:&nbsp;Any]</code> | No | Optional free-form data associated with this event. |
+| datastreamIdOverride | `String` | No | Optional Datastream identifier used to override the default datastream identifier set in the Edge configuration. |
+| datastreamConfigOverride | <code>[String:&nbsp;Any]</code> | No | Optional Datastream configuration used to override individual settings from the default datastream configuration. |
```
Comment (emdobrin) [inline Documentation/event-reference.md:93]: the event payload would contain config with the two new keys

File: Documentation/api-reference.md [hunk 4]
```diff
@@ -128,6 +132,69 @@ Edge.sendEvent(experienceEvent: experienceEvent) { (handles: [EdgeEventHandle])
 }
 ```
 
+##### Example with DatastreamIdOverride
```
Comment (emdobrin) [inline Documentation/api-reference.md:135]: nit: shall we use sentence like header: Example with Datastream ID override / Example with Datastream config override

File: Documentation/event-reference.md [hunk 5]
```diff
@@ -89,11 +89,24 @@ If the required `xdm` key is not present in the event data payload, the event is
 | --- | ---------- | -------- | ----------- |
 | xdm | <code>[String:&nbsp;Any]</code> | Yes | XDM formatted data; use an `XDMSchema` implementation for better XDM data ingestion and data format control. |
 | data | <code>[String:&nbsp;Any]</code> | No | Optional free-form data associated with this event. |
-| datastreamIdOverride | `String` | No | Optional Datastream identifier used to override the default datastream identifier set in the Edge configuration. |
-| datastreamConfigOverride | <code>[String:&nbsp;Any]</code> | No | Optional Datastream configuration used to override individual settings from the default datastream configuration. |
+| config | <code>[String:&nbsp;Any]</code> | No | Optional config overrides. |
```
Comment (emdobrin) [inline Documentation/event-reference.md:92]: nit: 
- Optional config settings. Find the available keys for `config` below.
- Optional request parameters. Find the available keys for `request` below.

File: Documentation/event-reference.md [hunk 6]
```diff
@@ -89,11 +89,24 @@ If the required `xdm` key is not present in the event data payload, the event is
...
 | datasetId | `String` | No | Optional custom dataset ID. If not set, the event uses the default Experience dataset ID set in the datastream configuration. |
 | request | <code>[String:&nbsp;Any]</code> | No | Optional request parameters. |
 
+config
+
+| Key | Value type | Required | Description |
+| --- | ---------- | -------- | ----------- |
+| config.datastreamIdOverride | `String` | No | Optional Datastream identifier used to override the default datastream identifier set in the Edge configuration. |
+| config.datastreamConfigOverride | <code>[String:&nbsp;Any]</code> | No | Optional Datastream configuration used to override individual settings from the default datastream configuration. |
```
Comment (emdobrin) [inline Documentation/event-reference.md:101]: - We should probably list just the inner key "datastreamIdOverride" / "datastreamConfigOverride" here, so users don't get confused that the key name is prefixed with "config."
- Same for request table

File: Documentation/api-reference.md [hunk 7]
```diff
@@ -100,13 +100,17 @@ See [MobileCore.resetIdentities](https://developer.adobe.com/client-sdks/documen
 
 Sends an Experience event to Edge Network.
 
+Starting with 4.3.0 the sentEvent API supports optional Datastream overrides. This allows you to adjust your datastreams without the need for new ones or modifications to existing settings. The process involves two steps:
```
Comment (cacheung) [inline Documentation/api-reference.md:103]: Can we be more specific about 4.3.0.
May be AEPEdge v4.3.0

**Review by emdobrin — APPROVED** [review]

**Pull request merged**